chat-plugins/panagrams-beta.js

```diff
@@ -152,4 +152,127 @@ exports.commands = {
 					"The type of this Pokemon is: <b>" + Tools.getTemplate(room.panagram.answer).types + "</b>";
 		this.sendReplyBox("Panagram hint: <br />The Pokemon is <b>" + answer_length + "</b> characters long.  <br />" + hint);
 	}
-};
\ No newline at end of file
+};*/
+
+//Panagrams by SilverTactic (Siiilver) and panpawn
+//#Tyme4BugHunting
+var pGames = Gold.pGames = {};
+var dex = Tools.data.pokedex;
+function mix(word) {
+	var arr = [];
+	for (var k = 0; k < word.length; k++) {
+		arr.push(word[k]);
+	}
+	var a, b, i = arr.length;
+	while (i) {
+		a = Math.floor(Math.random() * i);
+		i--;
+		b = arr[i];
+		arr[i] = arr[a];
+		arr[a] = b;
+	}
+	return arr.join('').toString();
+}
+
+var Panagram = (function () {
+	function Panagram(room, sessions) {
+		this.sessions = sessions;
+		this.room = room;
+		do {
+			this.answer = dex[Object.keys(dex)[Math.floor(Math.random() * Object.keys(dex).length)]];
+		} while (this.answer.num < 1 || this.answer.baseSpecies);
+		this.mixed = mix(this.category === 'pokemon' ? toId(this.answer.species) : this.answer.name.toLowerCase());
+
+		this.room.add('|html|<div class = "infobox"><center>A game of Panagram was started! Scrambled Pokemon: <b>' + this.mixed + '</b>.<br>' +
```

Comment:
Look at my ``broadcast-gold`` element I used for my current way of doing this.

Also, why not just ``this.add();``?